receiver/k8sclusterreceiver/collection/metadata.go

```diff
@@ -64,3 +63,93 @@ func getGenericMetadata(om *v1.ObjectMeta, resourceType string) *KubernetesMetad
 func getResourceIDKey(rType string) string {
 	return fmt.Sprintf("k8s.%s.uid", rType)
 }
+
+// mergeKubernetesMetadataMaps merges maps of string (resource id) to
+// KubernetesMetadata into a single map.
+func mergeKubernetesMetadataMaps(maps ...map[string]*KubernetesMetadata) map[string]*KubernetesMetadata {
+	out := map[string]*KubernetesMetadata{}
+	for _, m := range maps {
+		for id, km := range m {
+			out[id] = km
+		}
+	}
+
+	return out
+}
+
+// KubernetesMetadataExporter provides an interface to implement
+// ConsumeKubernetesMetadata in Exporters that support metadata.
+type KubernetesMetadataExporter interface {
+	ConsumeKubernetesMetadata(metadata map[string]*KubernetesMetadataUpdate) error
```

Comment:
Added comments on the method. Also, updated this method to take in a slice instead of map since we only really care about the updates themselves and don't need a mapping.